src/internal/m365/graph/uploadsession.go

```diff
@@ -75,5 +84,21 @@ func (iw *largeItemWriter) Write(p []byte) (int, error) {
 	// Update last offset
 	iw.lastWrittenOffset = endOffset
 
+	// Once the upload is complete, we get a Location header in the
+	// below format from which we can get the id of the uploaded
+	// item. This will only be available after we have uploaded the
+	// entire content(based on the size in the req header).
+	// https://outlook.office.com/api/v2.0/Users('<user-id>')/Messages('<message-id>')/Attachments('<attachment-id>')
+	loc := resp.Header.Get("Location")
+	if loc != "" {
+		splits := strings.Split(loc, "'")
```

Comment:
splitting on `'` feels pretty brittle. Can we guarantee they'll always quote values like that? Would it be better to split on `/` and then extract information from the final component? If they have a spec that states this then I'd be a bit more comfortable

Reply:
https://learn.microsoft.com/en-us/graph/outlook-large-attachments?tabs=http documents the format. I started of splitting by `/`, but then we will eventually have to split on `'`. Plus the checks [here](https://github.com/alcionai/corso/blob/01e10627630d7c20c3a46f43a9cd80f07fbc43d1/src/internal/m365/graph/uploadsession.go#L95) give me enough confidence.